src/Dict/Extra.elm

```diff
@@ -337,6 +383,31 @@ invert dict =
         dict
 
 
+{-| Like `invert`, it changes the keys and values. However, if one value maps to multiple keys, then all of the keys will be retained.
+
+    import Dict
+    import Set
+
+    Dict.fromList [ ( 1, "Jill" ), ( 2, "Jill" ), ( 3, "Jack" ) ]
+        |> invertAll
+    --> Dict.fromList [ ( "Jill", Set.fromList [ 1, 2 ] ), ( "Jack", Set.singleton 3 ) ]
+
+-}
+invertAll : Dict comparable1 comparable2 -> Dict comparable2 (Set comparable1)
+invertAll dict =
```

Comment:
I also feel like a (fuzz) test for this would be nice since it isn't completely trivial

Reply:
I ended up adding three :)